rts/lra/Dockerfile

```diff
@@ -0,0 +1,11 @@
+FROM fabric8/java-jboss-openjdk8-jdk:1.2.3
+
+ENV JAVA_APP_JAR lra-test-swarm.jar
+ENV AB_ENABLED off
+# to get more info about LRA processing in console you can add switch
+#-Dswarm.logging=TRACE
+ENV JAVA_OPTIONS -Xmx512m -Dlra.http.host=lra-coordinator-lra-minishift.192.168.99.100.nip.io -Dlra.http.port=80 -Dservice.http.host=trip-service-lra-minishift.192.168.99.100.nip.io -Dservice.http.port=80

```

Comment:
maybe I do not understand here the difference of the `lra/Dockerfile` and the `lra/lra-coordinator/Dockerfile`

but in generally we should not set here the address in form `lra-coordinator-lra-minishift.192.168.99.100.nip.io`, if we points to a service naming `lra-coordinator-lra-minishift` then we should use the in openshift resolution and set it just to `lra-coordinator-lra-minishift`

Reply:
If I change this I get:
```
2017-09-11 13:16:14,088 DEBUG [org.apache.http.impl.conn.BasicClientConnectionManager] (default task-1) Get connection for route {}->http://lra-coordinator-lra-minishift:80
--
  | 2017-09-11 13:16:14,106 DEBUG [org.apache.http.impl.conn.DefaultClientConnection] (default task-1) Connection org.apache.http.impl.conn.DefaultClientConnection@73de8cd2 closed
  | 2017-09-11 13:16:14,106 DEBUG [org.apache.http.impl.conn.DefaultClientConnection] (default task-1) Connection org.apache.http.impl.conn.DefaultClientConnection@73de8cd2 shut down
  | 2017-09-11 13:16:14,107 DEBUG [org.apache.http.impl.conn.BasicClientConnectionManager] (default task-1) Releasing connection org.apache.http.impl.conn.ManagedClientConnectionImpl@123281fa
```

Reply:
And if I don't include the lra-minishift: 

```[0m[32m2017-09-11 13:26:12,898 DEBUG [org.apache.http.impl.conn.DefaultClientConnectionOperator] (default task-2) Connecting to lra-coordinator:80
```

Reply:
But if I update the service to have 
```
-Dlra.http.host=lra-coordinator -Dlra.http.port=8080
```

Then I do get further in that the service can talk to the coordinator, however then in the coordinator log I get:
```
[0m[32m2017-09-11 13:29:34,511 DEBUG [org.apache.http.impl.conn.BasicClientConnectionManager] (default task-4) Get connection for route {}->http://lra-coordinator:8080
[0m[32m2017-09-11 13:29:34,527 DEBUG [org.apache.http.impl.conn.DefaultClientConnectionOperator] (default task-4) Connecting to lra-coordinator:8080
```

It's as if the coordinator decides it needs to talk to itself but can't access the right lookup...

Reply:
Similarly, the trip service has internal calls to itself and it is unable to resolve itself for those calls hanging at:

```

2017-09-11 13:42:06,584 DEBUG [org.apache.http.impl.conn.BasicClientConnectionManager] (default task-1) Get connection for route {}->http://trip-service:8080
--
  | 2017-09-11 13:42:06,586 DEBUG [org.apache.http.impl.conn.DefaultClientConnectionOperator] (default task-1) Connecting to trip-service:8080

```

Reply:
ah, ok, I see. This is the service Dockerfile. Then I hope that `-Dlra.http.host=lra-coordinator` should be set. Or explicitly use also `-Dlra.http.port=80` but the 80 will be probably the default port where to search the coordinator for.

Similar to the trip service - it should be available on port `80` (I hope)

Reply:
I agree - it should work. What I am trying to say is that the trip-service can have the lra-coordinator:8080 settings (-Dlra....) and talk to the coordinator but then the coordinator hangs trying to resolve *itself* on lra-coordinator:8080. The trip-service has similar problems as that needs to talk to itself also.

I haven't managed to work out why the lra-coordinator needs to talk to itself yet and I am going to split the services to different pods so that should help. Stay tuned!

Reply:
I agree - it should work. What I am trying to say is that the trip-service can have the lra-coordinator:8080 settings (-Dlra....) and talk to the coordinator but then the coordinator hangs trying to resolve itself on lra-coordinator:8080. The trip-service has similar problems as that needs to talk to itself also.

I haven't managed to work out why the lra-coordinator needs to talk to itself yet and I am going to split the services to different pods so that should help. Stay tuned!

Reply:
by interest - explicitly using `-Dlra.http.host=lra-coordinator -Dlra.http.port=80` would help? I mean the error message

```
DEBUG [org.apache.http.impl.conn.BasicClientConnectionManager] (default task-4) Get connection for route {}->http://lra-coordinator:8080
DEBUG [org.apache.http.impl.conn.DefaultClientConnectionOperator] (default task-4) Connecting to lra-coordinator:8080
```

talks about `lra-coordinator:8080`